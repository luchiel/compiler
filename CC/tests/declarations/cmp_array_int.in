void main()
{
float a[4];
int b;
a[0] == b;
}