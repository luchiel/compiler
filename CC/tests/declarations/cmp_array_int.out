type <int>
type <float>
type <void>
type <int*>
	pointer to <int>
type <void*>
	pointer to <void>
variable NULL of type <void*> initialized by
	{int 0}
		type <int>
function main
	returns type <void>
	arguments
		<no args>
	{{}}
		type <0>
			array [
				{int 4}
					type <int>
			] of <float>
		variable a of type <0>
		variable b of type <int>
	| 
	+-{;}
	  | 
	  +-{==}
				type <int>
	    | 
	    +-{[]}
					type <float>
	    | | 
	    | +-{a}
						type <0>
							array [
								{int 4}
									type <int>
							] of <float>
	    | | 
	    | +-{int 0}
						type <int>
	    | 
	    +-{cast}
					type <float>
	      | 
	      +-<float>
	      | 
	      +-{b}
						type <int>
